docs/sources/setup-grafana/configure-security/configure-authentication/generic-oauth/index.md

```diff
@@ -14,305 +14,194 @@ weight: 700
+| `tls_client_ca`              | No       | Path to the trusted certificate authority list.                                                                                                                                                |                 |
+| `use_pkce`                   | No       | Set to `true` to use "proof key for code exchange" (PKCE).                                                                                                                                     | `false`         |
+
+### Auth style
+
+`auth_style` controls which [OAuth2 AuthStyle](https://pkg.go.dev/golang.org/x/oauth2#AuthStyle) is used when token is requested from OAuth2 provider. It determines how `client_id` and `client_secret` are sent to Oauth2 provider.
+Available values are `AutoDetect`, `InParams` and `InHeader`. By default, `AutoDetect` is used.
 
-```bash
-[auth.generic_oauth]
-name = OAuth
-icon = signin
-enabled = true
-allow_sign_up = true
-auto_login = false
-client_id = YOUR_APP_CLIENT_ID
-client_secret = YOUR_APP_CLIENT_SECRET
-scopes =
-empty_scopes = false
-auth_url =
-token_url =
-api_url =
-allowed_domains = mycompany.com mycompany.org
-allowed_groups = ["Admins", "Software Engineers"]
-tls_skip_verify_insecure = false
-tls_client_cert =
-tls_client_key =
-tls_client_ca =
-use_pkce = true
-auth_style =
-```
+### Email address
 
-Set `api_url` to the resource that returns [OpenID UserInfo](https://connect2id.com/products/server/docs/api/userinfo) compatible information.
+Grafana determines a user's email address by following the steps below until it finds and e-mail address:
 
-You can also specify the SSL/TLS configuration used by the client.
+1. Check for the presence of an e-mail address via the `email` field encoded in the OAuth2 ID token.
+1. Check for the presence of an e-mail address using the [JMESPath](http://jmespath.org/examples.html) specified via the `email_attribute_path` configuration option. The JSON used for the path lookup is the HTTP response obtained from querying the UserInfo endpoint specified via the `api_url` configuration option.
+1. Check for the presence of an e-mail address in the `attributes` map encoded in the OAuth2 ID token. By default Grafana will perform a lookup into the attributes map using the `email:primary` key, however, this is configurable and can be adjusted by using the `email_attribute_name` configuration option.
+1. Query the `/emails` endpoint of the OAuth2 provider's API (configured with `api_url`), then check for the presence of an email address marked as a primary address.
+1. If no email address is found in steps (1-4), then the email address of the user is set to an empty string.
 
-- Set `tls_client_cert` to the path of the certificate.
-- Set `tls_client_key` to the path containing the key.
-- Set `tls_client_ca` to the path containing a trusted certificate authority list.
+### Login
```

Comment:
Same comment as with the email section. Why is this information important to the user? Seems like behind-the-scenes functioning of the system.